lib/puppet/provider/mysql.rb

```diff
@@ -24,7 +24,7 @@ def self.mysqld_type
     mysqld_version_string.scan(/mariadb/i) { return "mariadb" }
     mysqld_version_string.scan(/\s\(mysql/i) { return "mysql" }
     mysqld_version_string.scan(/\s\(percona/i) { return "percona" }
-    nil
+    return "mysql"
```

Comment:
I disagree. This is the mysql module, after all. mysql should be the default. Why nil? There are Ubuntu packages out there for mariadb as well.